torchao/quantization/GPTQ.py

```diff
@@ -1191,3 +1191,229 @@ def quantize(self, model: torch.nn.Module, inputs: List[_MultiInput], **kwargs:
         model = self._convert_for_runtime(model)
         model.load_state_dict(state_dict, strict=False)
         return model
+
+
+class SparseMarlinLinear(torch.nn.Module):
```

Comment:
You shouldn't need any of this @Diogo-V, see https://github.com/pytorch/ao/blob/main/torchao/quantization/GPTQ_MT.py#L619 you can use the tensor subclass directly, the only things you need to do are create the SparseMarlinGPTQQuantizer and then setup those functions i mentioned in my comment inside it

Reply:
Oh, you are right. I am going to refactor it